File: other/MT1051/week07/HW18/HW18_109601003.go
/*
Created Date: 2023/10/30
HW: 18
Author: 林群賀
Student Number: 109601003
*/

package main

import "fmt"

func funcValue(y float64) float64 {
	return (y + 3.2) * (y - 5.3) * (y + 2.7)
}

func main() {
	target := 1823.0
	lowerBound := 0.0
	upperBound := 20.0
	precision := 0.001

	for upperBound - lowerBound > precision {
		mid := (lowerBound + upperBound) / 2
		if funcValue(mid) < target {
			lowerBound = mid
		} else {
			upperBound = mid
		}
	}

	y := (lowerBound + upperBound) / 2

	if y > 12 {
		fmt.Println("y is greater than 12")
	} else if y < 12 {
		fmt.Println("y is less than 12")
	}
}

File: mt/mt1051/week07/HW18/HW18_109601003.py
# -*- coding: utf-8 -*-
"""
Date: 2023/10/30
HW: 18
Author: 林群賀
Student Number: 109601003
"""

def func(y):
    return (y + 3.2) * (y - 5.3) * (y + 9.2) * (y + 2.7)

target = 1823
lower_bound = 5.3
upper_bound = 20.0
precision = 0.001

while upper_bound - lower_bound > precision:
    mid = (lower_bound + upper_bound) / 2
    if func(mid) < target:
        lower_bound = mid
    else:
        upper_bound = mid

y = (lower_bound + upper_bound) / 2

if y > 12:
    print("y is greater than 12")
elif y < 12:
    print("y is less than 12")

